File: 0200-number-of-islands/0200-number-of-islands.java
class Solution {
    public int numIslands(char[][] grid) {
        int noOfIslands = 0;
        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(grid[i][j]=='1'){
                    noOfIslands++;
                    islandDFS(grid,i,j);
                }
            }
        }
        
        return noOfIslands;        
    }
    private void islandDFS(char[][] grid, int i, int j){
        if(i<0 || i>=grid.length || j<0 || j>=grid[0].length || grid[i][j] == '0')
            return;
        grid[i][j]='0';
        islandDFS(grid,i+1,j);
        islandDFS(grid,i-1,j);
        islandDFS(grid,i,j+1);
        islandDFS(grid,i,j-1);
    }
}

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        ans = 0
        def dfs(grid,i,j):
            if i<0 or j<0 or i>=len(grid) or j>= len(grid[0]) or \
            grid[i][j] == '0':
                return
            grid[i][j]='0'
            dfs(grid,i+1,j)
            dfs(grid,i-1,j)
            dfs(grid,i,j+1)
            dfs(grid,i,j-1)
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if(grid[i][j]=='1'):
                    ans+=1
                    dfs(grid,i,j)                    
        return ans
        